zebra-state/src/service/finalized_state/tests/prop.rs

```diff
@@ -0,0 +1,50 @@
+use std::env;
+
+use zebra_chain::{
+    block::{Block, Height, LedgerState},
+    parameters::Network,
+};
+use zebra_test::prelude::*;
+
+use crate::{
+    config::Config,
+    service::finalized_state::{FinalizedBlock, FinalizedState},
+};
+
+const DEFAULT_PARTIAL_CHAIN_PROPTEST_CASES: u32 = 2;
+const PARTIAL_CHAIN_SIZE: usize = 200;
```

Comment:
Suggestion: I'd probably try to weight this more towards cases - do you think anything interesting will happen after the first 10 or 20 blocks?

Cases generate new random chains, the chain size generates longer chains.

```suggestion
const DEFAULT_PARTIAL_CHAIN_PROPTEST_CASES: u32 = 32;
const PARTIAL_CHAIN_SIZE: usize = 10;
```